lua/plenary/path.lua

```diff
@@ -314,6 +314,47 @@ function Path:rmdir()
   uv.fs_rmdir(self:absolute())
 end
 
+function Path:rename(opts)
+  opts = opts or {}
+  if not opts.new_name or opts.new_name == "" then
+    print("Please provide the new name!")
+    return false
+  end
+
+  -- handles `.`, `..`, `./`, and `../`
+  if opts.new_name:match('^%.%.?/?') then
+    print("Invalid filename!")
+    return false
+  end
+
+  local new_file = self:new(opts.new_name)
+
+  if self:new(opts.new_name):exists() then
```

Comment:
I wouldn't use `self:new` i would use `Path:new` because self will call new on the current object and i am not sure about side effects when doing that. Never done this

Reply:
Also you create a new_file twice.